### PR TITLE
Replaced HoldString with RetainString for older Redis

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -1262,7 +1262,12 @@ static int DistSearchCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
   SearchCmdCtx* sCmdCtx = rm_malloc(sizeof(*sCmdCtx));
   sCmdCtx->argv = rm_malloc(sizeof(RedisModuleString*) * argc);
   for (size_t i = 0 ; i < argc ; ++i) {
+#ifdef HAVE_REDISMODULE_HOLDSTRING
     sCmdCtx->argv[i] = RedisModule_HoldString(ctx, argv[i]);
+#else
+    sCmdCtx->argv[i] = argv[i];
+    RedisModule_RetainString(ctx, argv[i]);
```
Comment (MeirShpilraien) [inline src/module.c:1269]: Why not just use retain string?

Comment (rafie) [reply src/module.c:1269]: I just wanted something to remind us the right way to go here. We can add a TODO comment if it helps.

Comment (MeirShpilraien) [reply src/module.c:1269]: I guess its also fine :+1:

**Review by MeirShpilraien — APPROVED** [review]

File: src/module.c (hunk 1 above)
Comment (ashtul) [reply src/module.c:1269]: We could use `isFeatureSupported` that we use to identify `RM_Scan`.

**Comment by MeirShpilraien** [comment]
@rafie we need to remember to add it to your PR that adds coordinator to RediSearch..

**Pull request merged**